### ISSUE TITLE
Add a Zoom mode key for pen & tablet use

### ISSUE THREAD

**Issue by iancustoica** [excalidraw/excalidraw#4626]
Hi,

I have started testing excalidraw for a bit in obsidian and I am looking for something like this: [Issue#794](https://github.com/excalidraw/excalidraw/issues/794) but for zoom. 
Something like while holding `ctrl+space` or `Z` and holding the left mouse button while dragging it up / down or left / right that would then have the same effect of `Ctrl + +` or `Ctrl + -` (zoom in / out).

This is very useful for people who use a pen & tablet such as myself, because I can activate it using the left hand on the keyboard and use the right hand with the pen, just like it works for panning (move canvas).

Thank you,
Iancu

**Comment by dwelle** [comment]
Pinch to zoom? But, once we add pen-mode where touch events will be ignore, that won't work. So a good point.

**Comment by Pedroespecial** [comment]
This is exactly what I am looking for I think.

In Adobe products I am able to hold <SPACE> to Pan the canvas and optionally hold <CMD> to Zoom with the pen.  This works so well for navigating a canvas.

I would also love to be able to add a modifier key to the <SPACE> to drag which allows me to Zoom with the Pen as happens currently when I <CMD +/->.

I'm using the Dial/Wheel of my Logitech Keyboard to Zoom in conjunction with <SPACE> just now, but it's a bit awkward.

Any ideas or hacks to achieve this would be most welcome :-)

Thank you
Pete

**Comment by dwelle** [comment]
> Weirdly, what I wanted just randomly happened! I was working with some text and my pen started zooming in and out! Unfortunately, I was stuck in this 'mode' and had to perform a Reload after which it all went back to how it normally is! Someone else mentioned this happening - https://github.com/excalidraw/excalidraw/pull/3931. It seems like this bug was fixed, but I wonder if there is a way to switch this 'bug' on and off?!

Mhm, so the bug wasn't fixed? 😭 The original PR you linked to didn't fix this fully, which we did in https://github.com/excalidraw/excalidraw/pull/4765 — but apparently not. Unless you were still on an older version, which can happen if you 1) didn't load Excalidraw in a while 2) you had multiple tabs open which prevents loading the latest version (ServiceWorker).

> In Adobe products I am able to hold to Pan the canvas and optionally hold to Zoom with the pen. This works so well for navigating a canvas.
> 
> I would also love to be able to add a modifier key to the to drag which allows me to Zoom with the Pen as happens currently when I <CMD +/->.
> 
> I'm using the Dial/Wheel of my Logitech Keyboard to Zoom in conjunction with just now, but it's a bit awkward.

We could maybe add support for various pen buttons (though I'm not sure the buttons are standardized so without having access to them it might be hard to do properly), but it will not solve this for cases like touchscreen tablets (iPad, Galaxy Note, ...).

**Comment by Pedroespecial** [comment]
1) The 'bug' only occurred once.  I am using ED in Obsidian with Zolts plugin.

2) I don't think any Pen button support is required.  It's kind of difficult to describe what I mean.... Currently, when I hold the Space bar down and pan up and down with my Pen I can move around the canvas...all good.  What I am also able to do on Adobe products and Krita is to additionally press the <CMD> key so <SPACE> + <CMD> and the same Pen panning action Zooms me in and out of the Canvas.  Thus by holding <SPACE> down and coming on and off of the <CMD> key I am able to Zoom **and** Pan around my canvas - allowing me super-quick control of where I want to next focus.

As a side note, I notice that I can also achieve a Zoom In and Out by holding down <CTRL> and using my Magic Mouse scroll, but sadly holding <CTRL> and Panning the Pen does not do the same.

**Comment by moonbyte13** [comment]
Is a pen-zoom feature planned for the future? Similar to the "space" to pan, but with zooming instead.

**Comment by JanitorHead** [comment]
Yeah, we need a analogous "space" to pan but with zooming. Concepts app does this with the "S" key, you can toggle it or hold it and then drag to zoom the canvas.

**Comment by MatheusMuriel** [comment]
There is a PR #7802 for this feature, but it seems to have been abandoned since last year. I am willing to take it on, I will make a new PR and submit it.

I hope it is accepted, as this feature would be of great help to anyone who uses a graphics tablet.

**Comment by MatheusMuriel** [comment]
@dwelle i want to pick it up. Can you assign it to me?
